swing-lib/src/main/java/org/triplea/swing/key/binding/SwingKeyBinding.java

```diff
@@ -0,0 +1,108 @@
+package org.triplea.swing.key.binding;
+
+import java.awt.KeyboardFocusManager;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.text.JTextComponent;
+import lombok.experimental.UtilityClass;
+import org.triplea.swing.SwingAction;
+
+/**
+ * Supports adding key bindings. Reference:
+ * https://docs.oracle.com/javase/tutorial/uiswing/misc/keybinding.html
+ *
+ * <p>Key bindings added through this class will "always" be active regardless if the component is
+ * currently focused or not. Single key key bindings will not be active when focus is inside of a
+ * text component. This is to avoid, for example, a user typing a chat message and unintentionally
+ * firing key bindings.
+ */
+@UtilityClass
+public class SwingKeyBinding {
+  public static void addKeyBinding(
+      final JDialog frame, final KeyCombination keyCombination, final Runnable action) {
+    addKeyMapping(frame.getRootPane(), keyCombination, action);
+  }
+
+  public static void addKeyBinding(
+      final JFrame frame, final KeyCombination keyCombination, final Runnable action) {
+    addKeyMapping(frame.getRootPane(), keyCombination, action);
+  }
+
+  public static void addKeyBinding(
+      final JDialog frame, final KeyCode keyCode, final Runnable action) {
+    addKeyMapping(frame.getRootPane(), keyCode, action);
+  }
+
+  public static void addKeyBinding(
+      final JFrame frame, final KeyCode keyCode, final Runnable action) {
+    addKeyMapping(frame.getRootPane(), keyCode, action);
+  }
+
+  public static void addKeyBinding(
+      final JComponent component, final KeyCode keyCode, final Runnable action) {
+    addKeyMapping(component, keyCode, action);
+  }
+
+  /**
+   * Adds two keybindings to a given key, both active when "ctrl" and "meta" are held down.
+   *
+   * @param key The key to be bind in combination with "ctrl" being held down, or "meta" being held
+   *     down.
+   * @param action The action to execute when the keybinding is activated.
+   */
+  public static void addKeyListenerWithMetaAndCtrlMasks(
+      final JFrame component, final KeyCode key, final Runnable action) {
+
+    addKeyMapping(
+        (JComponent) component.getContentPane(),
+        KeyCombination.of(key, ButtonDownMask.CTRL),
+        action);
+    addKeyMapping(
+        (JComponent) component.getContentPane(),
+        KeyCombination.of(key, ButtonDownMask.META),
+        action);
+  }
+
+  private static void addKeyMapping(
+      final JComponent component, final KeyCode keyCode, final Runnable action) {
+    addKeyMapping(component, KeyCombination.of(keyCode, ButtonDownMask.NONE), action);
```

Comment:
What do you think about making KeyCombination and KeyCode implement the same interface?
This way you woldn't need to convert one to the other and were able to nest things in theory

Reply:
Not a  bad thought given the common method. It does not help that much though as we need to convert `KeyCode` into `KeyCombination`.

Reply:
Going along those lines, `KeyCode` actually does not need the `toKeyStroke` at all and can be removed.